proofs/tree.go

```diff
@@ -312,6 +318,13 @@ func (doctree *DocumentTree) AddLeaf(leaf LeafNode) error {
 	if doctree.filled {
 		return errors.New("tree already filled")
 	}
+	for _, item := range doctree.leaves {
```

Comment:
@vedhavyas @lucasvo 
as far as performance is concerned, my suggestion is :

introducing 
```
addLeave(leaf LeafNode, checkDuplicated bool)
addLeaves(leaves []LeafNode, checkDuplicated bool)
```
when client  call `AddLeavesFromDocument
`, then by pass duplication checking, because protocol buffer enforce the checking (I think this is the main usage scenario?)
if client call `AddLeave `or `AddLeaves `directly, then make duplication checking. (little used scenarion?)

In this way, we need not keep Leaf adding status (two sets or maps)  in `DocumentTree `

What's your opinions?

Reply:
@zyfrank, I think we want to enforce a duplicate check. You don't need to add an option to skip it. AddLeavesFromDocument shouldn't skip it because you could accidentally add two messages of the same messages to the tree which would also lead to a duplicate.

Reply:
@lucasvo ,you are right. 
so we  keep Leaf adding status (two sets or maps, one for readable name, another for compact) in DocumentTree to improve performance ?

Reply:
Yeap. One for Name and one for Property should be good